splitio/models/grammar/matchers/rule_based_segment.py

```diff
@@ -29,20 +30,45 @@ def _match(self, key, attributes=None, context=None):
         if self._rbs_segment_name == None:
             return False
         
-        # Check if rbs segment has exclusions
-        if context['ec'].segment_rbs_memberships.get(self._rbs_segment_name):
+        rb_segment = context['ec'].rbs_segments.get(self._rbs_segment_name)
+        
+        if key in rb_segment.excluded.get_excluded_keys():
+            return False 
+
+        if self._match_dep_rb_segments(rb_segment.excluded.get_excluded_segments(), key, attributes, context):
             return False
                 
-        for parsed_condition in context['ec'].segment_rbs_conditions.get(self._rbs_segment_name):
-            if parsed_condition.matches(key, attributes, context):
-                return True
-                
-        return False
+        return self._match_conditions(rb_segment.conditions, key, attributes, context)
     
     def _add_matcher_specific_properties_to_json(self):
         """Return UserDefinedSegment specific properties."""
         return {
             'userDefinedSegmentMatcherData': {
                 'segmentName': self._rbs_segment_name
             }
-        }
\ No newline at end of file
+        }
+        
+    def _match_conditions(self, rbs_segment_conditions, key, attributes, context):
+        for parsed_condition in rbs_segment_conditions:
+            if parsed_condition.matches(key, attributes, context):
+                return True
+            
+        return False
+            
+    def _match_dep_rb_segments(self, excluded_rb_segments, key, attributes, context):
+        for excluded_rb_segment in excluded_rb_segments:
+            if excluded_rb_segment.type == SegmentType.STANDARD:
+                if context['ec'].segment_memberships[excluded_rb_segment.name]:
+                    return True
+            else:
+                excluded_segment = context['ec'].rbs_segments.get(excluded_rb_segment.name)
+                if key in excluded_segment.excluded.get_excluded_keys():
+                    return False
```

Comment:
why this return `false`?

Reply:
It matches to false if excluded keys contain the matching key